Modules/rat_facts.py

```diff
@@ -0,0 +1,187 @@
+class Fact:
+    name = None
+    language = None
+    message = None
+    author = None
+    timestamp = None
+
+    def __init__(self, name: str, lang: str, message: str, author: str,
+                 timestamp: datetime.datetime):
+        self.name = name
+        self.lang = lang
+        self.message = message
+        self.author = author
+        self.timestamp = timestamp
+
+
+class FactsManager(Singleton):
+
+    @property
+    def enabled(self):
+        return self.dbm.enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        raise RuntimeError("You cant set this. Really. If you want me disable, disable the DBM.")
+
+    def __init__(self):
+        """
+        retrieves the DBM object.
+        """
+        self.dbm = DatabaseManager()
+
+    async def get_fact(self, name: str, lang: str = "en") -> Fact or None:
+        """
+        retrieves a fact from the DB.
+        Args:
+            name: name of the fact to retrieve
+            lang: language to retrieve
+
+        Returns: message of the fact, or None should it not be found for the specified language
+
+        """
+        result = await self.dbm.select_rows("fact", "AND", {"lang": lang, "name": name})
+        timestamp = (await self.dbm.select_rows("fact_timestamps", "AND",
+                                                {"lang": lang, "name": name}))
+        if len(timestamp) > 0:
+            timestamp = timestamp[0].last_modified
+        else:
+            timestamp = datetime.datetime.utcfromtimestamp(0)
+        if result:
+            result = result[0]
+            return Fact(
+                result[0],
+                result[1],
+                result[2],
+                result[3],
+                timestamp
+            )
+        else:
+            return None
+
+    async def is_fact(self, name: str, lang: str = "en") -> bool or None:
+        """
+        checks for existence of a fact
+        Args:
+            name: name of the fact to check
+            lang: language to check
+
+        Returns: True if it exists, False if it exists
+
+        """
+        result = await self.dbm.select_rows("fact", "AND", {"lang": lang, "name": name})
+        return len(result) > 0
+
+    async def set_fact(self, fact: Fact) -> bool or None:
+        """
+        sets a fact
+        Args:
+            fact (Fact): the fact to push into the DB
+        Returns: True on success, False if it failed
+
+        """
+        try:
+            await self.dbm.insert_row("fact",
+                                      {"name": fact.name,
+                                       "lang": fact.lang,
+                                       "message": fact.message,
+                                       "author": fact.author},
+                                      ("name", "lang",)
+                                      )
+        except ValueError:
+            return False
+        else:
+            time = datetime.datetime.now(tz=datetime.timezone.utc)
```

Comment:
`datetime.datetime.utcnow`?

Reply:
no, utcnow gives timezone-naive not timezone-aware datetimes.
Sadly.

Reply:
ah. darn.

Reply:
funny enough we encountered the problems with `datetime.utcnow` in mecha2 as well. Caused a major problem with trusting times set by mecha in the quote object.

Reply:
Yeah. i remember fixing that too. In one place timezone aware timestamps were written but in another timezone-naive. What we should probably do is have a sanity check here to ensure that the data is correctly formatted prior to interpreting it